src/components/Menu.vue

```diff
@@ -0,0 +1,178 @@
+<template>
+  <popover :visible="visible" :align="alignLeft" bottom @closed="$emit('closed')">
+    <div class="menu__body">
+      <transition name="slide-left" mode="out-in">
+        <div v-if="visiblePanel == 1">
+          <div class="menu__panel">
+            <div
+              v-for="button in buttons"
+              :key="button.label"
+              class="menu__option"
+              @click="button.onclick"
+              v-html="button.html"
```

Comment:
I think it's best to avoid html if we can.
Here where labels seems enough :)

Reply:
Actually I realized you probably did that for data types icons.
I think it would even be better to have a slot here and a MenuButton component, so that the usage looks lioke:
```
<Menu>
  <MenuButton @click="...">Some operation</MenuButton>
  <MenuButton @click="...">Some operation</MenuButton>
  ...
</Menu>
```
It seems to me that this kind of declarative API, based on templates, is easier to read, understand and extend, that the array-based one.

Reply:
I took a minute to check on Vuetify's approach, and they also use this kind of template API. See https://vuetifyjs.com/en/components/lists/